### ISSUE TITLE
Misleading error message when actication key for systemd-cryptsetup is invalid

### ISSUE THREAD

**Issue by kbabioch** [systemd/systemd#11498]
**systemd version the issue has been seen with**
> 240.34

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > Error message that tells me that the key is invalid

**Unexpected behaviour you saw**
 > Error message that mislead me because of an indication that it is related to permissions: `Operation not permitted`

```
Jan 17 18:28:29 archlinux systemd-cryptsetup[295]: Failed to activate with key file '/etc/abc123': Operation not permitted
```

The permissions on this file are fine, though and if the file contains the correct key, it works just fine. However if the file contains the wrong key (e.g. due to a trailing new line), it the error message might mislead the administrator that there is an issue with file permissions.

**Steps to reproduce the problem**
Reference a key file in `/etc/crypttab` that is used during early boot to unlock a device.

**Comment by poettering** [comment]
hmm, that's the error code we get from the kernel there, and we currently output it 1:1 on modified. I figure we should suppress that though and write something smarter there instead.

**Comment by poettering** [comment]
I prepped a fix in #11513. Could you give it a whirl?
